tests/qactivation_test.py

```diff
@@ -301,6 +302,45 @@ def test_quantized_sigmoid(bits, sigmoid_type, use_real_sigmoid, test_values, ex
   assert_allclose(result, expected_values, rtol=1e-05)
 
 
+@pytest.mark.parametrize(
+    'bits, use_real_tanh, test_values, expected_values', [
+        (
+            4,
+            False,
+            np.array(
+                [[-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75]],
+                dtype=K.floatx()),
+            np.array([[-0.875, -0.75 , -0.5  , -0.25 ,  0.   ,  0.25 ,  0.5  ,  0.75 ]],
+                     dtype=K.floatx()),
+        ),
+        (
+            4,
+            True,
+            np.array(
+                [[-1.  , -0.75, -0.5 , -0.25,  0.  ,  0.25,  0.5 ,  0.75]],
+                dtype=K.floatx()),
+            np.array([[-0.75 , -0.625, -0.5  , -0.25 ,  0.   ,  0.25 ,  0.5  ,  0.625]],
+                     dtype=K.floatx()),
+        )
+    ])
+def test_quantized_tanh(bits, use_real_tanh, test_values, expected_values):
+  """Test quantized_sigmoid function with three different sigmoid variants."""
+  # store previous sigmoid type
+  if quantized_sigmoid(5)(K.cast_to_floatx([1.0])).numpy()[0] == 0.96875:
```

Comment:
This is not a reliable way to test what the previous sigmoid type was since it is reliant on the testing input. I suggest this approach instead:

In the `quantizers.py` file, set a default sigmoid type on line 204 like so:
```
_default_sigmoid_type = "hard"
_sigmoid = None


def set_internal_sigmoid(mode):
  """Sets _sigmoid to either real, hard or smooth."""

  global _sigmoid

  if mode not in ["real", "hard", "smooth"]:
    raise ValueError("mode has to be 'hard' or 'smooth'.")

  if mode == "hard":
    _sigmoid = hard_sigmoid
  elif mode == "smooth":
    _sigmoid = smooth_sigmoid
  elif mode == "real":
    _sigmoid = tf.keras.backend.sigmoid

set_internal_sigmoid(_default_sigmoid_type)
```

Then in `qactivation_test.py`, you can simply import `_default_sigmoid_type` and set the sigmoid type to that at the end of every test like so:
```
set_internal_sigmoid(_default_sigmoid_type)
```